openedx/core/djangoapps/site_configuration/middleware.py

```diff
@@ -38,6 +38,10 @@ def _set_cookie_wrapper(key, value='', max_age=None, expires=None, path='/', dom
                 if key == configuration_helpers.get_value('SESSION_COOKIE_NAME', settings.SESSION_COOKIE_NAME):
                     domain = session_cookie_domain
 
+                # Override the domain when the cookie is the LANGUAGE_COOKIE.
+                elif key == settings.LANGUAGE_COOKIE:
+                    domain = configuration_helpers.get_value('LANGUAGE_COOKIE_DOMAIN', session_cookie_domain)
+
```

Comment:
```suggestion
                language_cookie_domain = configuration_helpers.get_value('LANGUAGE_COOKIE_DOMAIN')
                
                elif key == settings.LANGUAGE_COOKIE and language_cookie_domain:
                    domain = language_cookie_domain
```
@Jacatove What do you think? It's to avoid any relationship between LANGUAGE_COOKIE_DOMAIN and SESSION_COOKIE_NAME.